src/trusted/service_runtime/nacl_app_thread.c

```diff
@@ -146,12 +146,13 @@ struct NaClApp *NaClChildNapCtor(struct NaClApp *nap) {
   
   /* duplicate file descriptor table starting at child_fd = 3 (0-2 setup previously)*/
   NaClXMutexLock(&nap_parent->mu);
-  for (int parent_fd = nap_child->fd; parent_fd <= nap_parent->fd; parent_fd++) {
+
+  for (int fd = 0; fd <= FILE_DESC_MAX; fd++) {
 
     /* Retrive the host fd we had stored in the Cage Table for the parent */
-    int parent_host_fd = fd_cage_table[nap_parent->cage_id][parent_fd];
-    if (parent_host_fd == 0) {
-      fd_cage_table[nap_child->cage_id][nap_child->fd++] = 0;
+    int parent_host_fd = fd_cage_table[nap_parent->cage_id][fd];
+    if (parent_host_fd == -1) {
```

Comment:
I presume this is being set so that you don't confuse it with stdin.  Would a special #defined macro value be better?
```suggestion
    // Using -1 (an invalid fd) to avoid confusing this with a valid fd.
    if (parent_host_fd == -1) {
```

Reply:
A macro definitely seems like a good idea!